### PR TITLE
[ASAN] Fix AddressSanitizer heap-buffer-overflow

### LINKED ISSUE #32896: L1 TrackQuality::featureTransform reads off the end of vector
The ASAN report shows that the function TrackQuality::featureTransform  is reading off the end of a vector.

### PR DESCRIPTION
**Pull request by smuzaffar**
This fixes the head buffer overflow issue seen in ASAN IBs https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc900/CMSSW_11_3_ASAN_X_2021-02-15-2300/pyRelValMatrixLogs/run/29834.0_TTbar_14TeV+2026D64+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal/step2_TTbar_14TeV+2026D64+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal.log#/

Fixes https://github.com/cms-sw/cmssw/issues/32896

[a] In ASAN IB , after the fix
```
29834.0_TTbar_14TeV+2026D64+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Tue Feb 16 21:41:44 2021-date Tue Feb 16 21:05:48 2021; exit: 0 0 0 0
1 1 1 1 tests passed, 0 0 0 0 failed

```

### CODE REVIEW

File: L1Trigger/TrackTrigger/src/TrackQuality.cc [hunk 1]
```diff
@@ -90,13 +90,13 @@ std::vector<float> TrackQuality::featureTransform(TTTrack<Ref_Phase2TrackerDigi_
...
+  for (int j = 1; j < eta_size; j++) {
+    if (eta < eta_bins[j] && eta >= eta_bins[j - 1])  // if track in eta bin
     {
       // Iterate through hitpattern binary
       for (int k = 0; k <= 6; k++)
         // Fill expanded binary entries using the expected hitmap table positions
-        hitpattern_expanded_binary[hitmap[j][k]] = hitpattern_binary[k];
+        hitpattern_expanded_binary[hitmap[j-1][k]] = hitpattern_binary[k];
     }
   }
```
Comment (smuzaffar) [inline L1Trigger/TrackTrigger/src/TrackQuality.cc:101]: @cms-sw/l1-l2 I personally world like to change it to 
```
-  for (int j = 0; j < eta_size; j++) {
-    if (eta >= eta_bins[j] && eta < eta_bins[j + 1])  // if track in eta bin
-    {
-      // Iterate through hitpattern binary
-      for (int k = 0; k <= 6; k++)
-        // Fill expanded binary entries using the expected hitmap table positions
-        hitpattern_expanded_binary[hitmap[j][k]] = hitpattern_binary[k];
-    }
+  for (int j = 1; j < eta_size; j++) {
+    if (eta >= eta_bins[j])
+        continue;
+    // Iterate through hitpattern binary
+    for (int k = 0; k <= 6; k++)
+      // Fill expanded binary entries using the expected hitmap table positions
+      hitpattern_expanded_binary[hitmap[j-1][k]] = hitpattern_binary[k];
+    break;
   }
```

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by skinnari** [comment]
@Chriisbrown @tomalin FYI

**Comment by makortel** [comment]
@cms-sw/l1-l2 @cms-sw/upgrade-l2 Could you please take a look?

**Comment by rekovic** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**